services/rollout-service/pkg/versions/versions.go

```diff
@@ -72,6 +77,67 @@ func (v *versionClient) GetVersion(ctx context.Context, revision, environment, a
 	return 0, nil
 }
 
+type KuberpultEvent struct {
+	Environment string
+	Application string
+	Version     uint64
+}
+
+type VersionEventProcessor interface {
+	ProcessKuberpultEvent(ctx context.Context, ev KuberpultEvent)
+}
+
+func (v *versionClient) Subscribe(ctx context.Context, processor VersionEventProcessor) error {
+	ctx = auth.WriteUserToGrpcContext(ctx, RolloutServiceUser)
+outer:
+	for {
+		client, err := v.client.StreamOverview(ctx, &api.GetOverviewRequest{})
+		if err != nil {
+			logger.FromContext(ctx).Warn("overview.connect", zap.Error(err))
+			continue outer
+		}
+	inner:
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			default:
+			}
+			overview, err := client.Recv()
+			if err != nil {
+				grpcErr := grpc.UnwrapGRPCStatus(err)
+				if grpcErr != nil {
+					if grpcErr.Code() == codes.Canceled {
+						return nil
+					}
+
+					logger.FromContext(ctx).Warn("overview.stream", zap.Error(err), zap.String("grpc.code", grpcErr.Code().String()), zap.String("grpc.message", grpcErr.Message()))
+				} else {
+					logger.FromContext(ctx).Warn("overview.stream", zap.Error(err))
+				}
+				continue inner
```

Comment:
wouldn't just `continue` do the same thing here?

Reply:
Yes. Done as well.